compiler/rustc_parse/src/parser/path.rs

```diff
@@ -624,9 +624,18 @@ impl<'a> Parser<'a> {
             GenericArg::Const(self.parse_const_arg()?)
         } else if self.check_type() {
             // Parse type argument.
+            let is_const_fn = self.look_ahead(1, |t| t.kind == token::OpenDelim(token::Paren));
+            let mut snapshot = self.clone();
```

Comment:
I'm a little hesitant that we're cloning the parser before we know whether or not there is an error, but it looks like we do this elsewhere so I guess it's fine.